src/generic/numberparsing.h

```diff
@@ -521,21 +1431,58 @@ really_inline bool parse_number(UNUSED const uint8_t *const src,
         // Ok, chances are good that we had an overflow!
         // this is almost never going to get called!!!
         // we start anew, going slowly!!!
-        return parse_float(src, parser, found_minus);
+        double d;
+        if(parse_float_strtod((const char*)src, &d)) {
+          parser.on_number_double(d);
+#ifdef JSON_TEST_NUMBERS // for unit testing
+          found_float(d, src);
+#endif
+          return true;
+        }
+#ifdef JSON_TEST_NUMBERS // for unit testing
+        found_invalid_number(src);
+#endif
+        return false;
       }
     }
-    if (unlikely((power_index > 2 * 308))) { // this is uncommon!!!
+    if (unlikely(exponent < FASTFLOAT_SMALLEST_POWER) ||
+      (exponent > FASTFLOAT_LARGEST_POWER)) {// this is uncommon!!!
       // this is almost never going to get called!!!
       // we start anew, going slowly!!!
-      return parse_float(src, parser, found_minus);
+        double d;
+        if(parse_float_strtod((const char *)src, &d)) {
+          parser.on_number_double(d);
+#ifdef JSON_TEST_NUMBERS // for unit testing
+          found_float(d, src);
+#endif
+          return true;
+        }
+#ifdef JSON_TEST_NUMBERS // for unit testing
+        found_invalid_number(src);
+#endif
+        return false;
     }
-    double factor = power_of_ten[power_index];
-    factor = negative ? -factor : factor;
-    double d = i * factor;
-    parser.on_number_double(d);
+    double d = 0;
+    bool success = true;
+    if (i != 0) {
```

Comment:
Is it worth the if statement here? It seems like this will only happen for weird expressions like `0e10` or `0.000`. I would guess it happens seldom enough that you don't want to waste a single instruction on it--just let the `0 * x` or `0 / x` happen.

Reply:
Yes. I wondered the same thing. To be tested. If the speed was the same, it would be worth it to remove it just to simplify the code.

Reply:
i think it may have been causing logic errors for me without it

Reply:
It is still needed, but I have now hidden it better.